contrib/golang/filters/http/source/config.cc

```diff
@@ -0,0 +1,55 @@
+#include "contrib/golang/filters/http/source/config.h"
+
+#include "envoy/registry/registry.h"
+
+#include "source/common/common/fmt.h"
+
+#include "contrib/golang/filters/http/source/common/dso/dso.h"
+#include "contrib/golang/filters/http/source/golang_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Golang {
+
+Http::FilterFactoryCb GolangFilterConfig::createFilterFactoryFromProtoTyped(
+    const envoy::extensions::filters::http::golang::v3alpha::Config& proto_config,
+    const std::string&, Server::Configuration::FactoryContext& context) {
+
+  FilterConfigSharedPtr config = std::make_shared<FilterConfig>(proto_config);
+
+  ENVOY_LOG_MISC(info, "load golang library at parse config: {} {}", config->soId(),
+                 config->soPath());
+
+  auto res = Envoy::Dso::DsoInstanceManager::load(config->soId(), config->soPath());
```

Comment:
Shouldn't we be unloading when the last reference to this by a filter is dropped otherwise we might leak open handles? Maybe that doesn't matter in the first version but I would at least add a TODO/comment about this.

Reply:
Thanks for your comments. Because the type of `DsoInstancePtr` is an `shared_ptr` and it will be destructed when there is no reference.  As below

https://github.com/envoyproxy/envoy/blob/00602640553c04f82cf6b8f78cf96f2563bc9479/contrib/golang/filters/http/source/common/dso/dso.cc#L82-L84

Reply:
The problem is you don't capture res, so I don't understand how this stays alive?

Reply:
> The problem is you don't capture res, so I don't understand how this stays alive?

@mattklein123. if the `load`  return failed then the DSO instance don't store the `dsoStore` map and close the handle in the destructor. otherwise store the DSO instance shared ptr to the `dsoStore` map for keepalive. As below

https://github.com/envoyproxy/envoy/blob/00602640553c04f82cf6b8f78cf96f2563bc9479/contrib/golang/filters/http/source/common/dso/dso.cc#L16-L21

Reply:
Yes but when does it ever get deleted? That's my point.

Reply:
> Yes but when does it ever get deleted? That's my point.

@mattklein123  The type of `DsoInstancePtr` is `std::shared_ptr<DsoInstance>` and it will be automatically destructed/deleted when there is no reference. 

https://github.com/envoyproxy/envoy/blob/00602640553c04f82cf6b8f78cf96f2563bc9479/contrib/golang/filters/http/source/common/dso/dso.h#L47

Reply:
You store it in a static. When is it ever deleted? Also now that I look I don't see unload() ever called. AFAICT you are leaking DSO loads if a filter gets loaded and unloaded.

Reply:
Yes. I see what you mean and i'll add some comments for it.

Reply:
Done. thanks.